robot_calibration/src/chain_manager.cpp

```diff
@@ -39,33 +39,38 @@ ChainManager::ChainManager(ros::NodeHandle& nh, double wait_time)
   for (int i = 0; i < chains.size(); ++i)
   {
     std::string name, topic, group;
+    bool has_controller;
     name = static_cast<std::string>(chains[i]["name"]);
     topic = static_cast<std::string>(chains[i]["topic"]);
     group = static_cast<std::string>(chains[i]["planning_group"]);
+    has_controller = static_cast<bool>(chains[i]["has_controller"]);
```

Comment:
rather than adding another parameter, we could do this by not setting the "topic" parameter (since, if there is no controller, there will be no topic). You can then use chains[i].hasMember("topic") to decide if we should create a controller